stable/gocd/templates/gocd-server-deployment.yaml

```diff
@@ -24,6 +24,10 @@ spec:
         app: {{ template "gocd.name" . }}
         release: {{ .Release.Name | quote }}
         component: server
+      annotations:
```

Comment:
should we also consider setting these annotations on deployment instead of pod template?

Reply:
Hi, we didn't think of allowing annotations on deployment because the use case we were thinking about was to allow the pod to assume roles via the kube2iam library.

Reply:
In that sense, can we add 2 kinds of annotations?

one for deployment, another for the pod?

Reply:
the values under values.yaml will look like:

`server.deployment.annotations` and `server.pod.annotations`?

Reply:
Ok, sounds like a good idea